### PR TITLE
Split authentication and tokeninfo functionality

### PR DESCRIPTION
**Pull request by cberg-zalando**
To be able to use both modules separately,I declared the specific modules as public and also splitted the settings classes for both of them.

### CODE REVIEW

File: src/main/java/org/zalando/undertaking/oauth2/OAuth2Settings.java [hunk 1]
```diff
@@ -6,7 +6,12 @@
 
 /**
  * Provides configuration settings for the OAuth2 functionality.
+ *
+ * @deprecated  Use {@link AccessTokenSettings} and {@link AuthenticationInfoSettings} directly. This avoids having to
+ *              implement methods, which are not required when only one of the two settings are required. This class is
+ *              scheduled for removal in Undertaking 0.1.0.
  */
+@Deprecated
```
Comment (twz123) [inline src/main/java/org/zalando/undertaking/oauth2/OAuth2Settings.java]: Maybe you should extend from the other settings here?

**Comment by twz123** [comment]
:+1:

File: src/main/java/org/zalando/undertaking/oauth2/OAuth2Settings.java (hunk 1 above)
Comment (cberg-zalando) [inline src/main/java/org/zalando/undertaking/oauth2/OAuth2Settings.java]: Good idea. So much for `DRY`. :smile:

**Comment by cberg-zalando** [comment]
:+1:

**Comment by twz123** [comment]
:+1:

**Pull request merged**